packages/mui-system/src/ThemeProvider/ThemeProvider.js

```diff
@@ -64,12 +64,15 @@ function ThemeProvider(props) {
   const engineTheme = useThemeScoping(themeId, upperTheme, localTheme);
   const privateTheme = useThemeScoping(themeId, upperPrivateTheme, localTheme, true);
   const rtlValue = engineTheme.direction === 'rtl';
-
   return (
     <MuiThemeProvider theme={privateTheme}>
       <StyledEngineThemeContext.Provider value={engineTheme}>
         <RtlProvider value={rtlValue}>
-          <DefaultPropsProvider value={engineTheme?.components}>{children}</DefaultPropsProvider>
+          <DefaultPropsProvider
+            value={themeId ? engineTheme[themeId].components : engineTheme.components}
```

Comment:
Are there other places we need to check for `themeId`? For example

```js
  const rtlValue = themeId ? engineTheme[themeId].direction === 'rtl' : engineTheme.direction === 'rtl';
```

Or

```jsx
<StyledEngineThemeContext.Provider value={themeId ? engineTheme[themeId] : engineTheme}>
```

Or is it only necessary for the components property?

Reply:
good call. updated the `rtlValue` to use `engineTheme[themeId]` if necessary.